### PR TITLE
Update Glimmer

### PR DESCRIPTION
**Pull request by chadhietala**
This commit brings glimmer `master` and ember `master` in line.

### CODE REVIEW

File: packages/ember-glimmer/lib/environment.js [hunk 1]
```diff
@@ -180,16 +180,13 @@ export default class Environment extends GlimmerEnvironment {
     }
   }
 
-  hasHelper(nameParts, symbolTable) {
-    assert('The first argument passed into `hasHelper` should be an array', Array.isArray(nameParts));
+  hasHelper(name, symbolTable) {
 
     // helpers are not allowed to include a dot in their invocation
-    if (nameParts.length > 1) {
+    if (name.indexOf('.') > 1) {
```
Comment (chancancode) [inline packages/ember-glimmer/lib/environment.js]: ✂️  I took care of this in the pre-compiler already

File: packages/ember-glimmer/lib/environment.js [hunk 2]
```diff
@@ -239,31 +233,27 @@ export default class Environment extends GlimmerEnvironment {
...
     }
   }
 
-  hasModifier(nameParts) {
-    assert('The first argument passed into `hasModifier` should be an array', Array.isArray(nameParts));
-
+  hasModifier(name) {
     // modifiers are not allowed to include a dot in their invocation
-    if (nameParts.length > 1) {
+    if (name.indexOf('.') > -1) {
```
Comment (chancancode) [inline packages/ember-glimmer/lib/environment.js]: ✂️  I took care of this in the pre-compiler already

**Review by chancancode — APPROVED** [review]

**Comment by rwjblue** [comment]
Need some more cases for the AST transform:

```hbs 
{{foo.bar bar="something"}}
```

```hbs
{{foo.bar derp}}
```

```hbs 
{{#foo.bar}}stuff here{{/foo.bar}}
```

Specifically, rendering previously saved off closure components that are not derived from block params is totally valid and should be supported.

**Comment by rwjblue** [comment]
👏

**Pull request merged**